tracks/python/exercises/acronym/mentoring.md

```diff
@@ -0,0 +1,75 @@
+### Problem and challenges
+
+The problem requires the student to split a string at word boundaries,
+and select the first letter of each word.
+
+### Reasonable solutions
+
+There are two characters that require special attention: 
+the hypen, used to tie together words such as "metal-oxide", 
+and the underscore, which must not be included.  
+The underscore can be silently remmoved, but the hypen marks 
+the boundary between two words, and it is simplest to replace 
+it with a space.
+
+```python
+    def abbreviate(words):
+        words = words.upper()
+        words = words.replace('_', '')      # Remove underscore
+        words = words.replace('-', ' ')     # Transform hyphen
+
+        tla = [ word[0] for word in words.split(' ') if word ]
+        return ''.join(tla)
+```
+
+#### Assembling Strings
+
+Some solutions will assemble the result string one character at a time:
+
+```python
+    def abbreviate(words):
+        words = words.replace('_', '')
+        words = words.replace('-', ' ')
+
+        word_list = words.split(' ')
+    
+        tla = ""
+        for word in word_list:
+            if word:
+                tla = tla + word[0].upper()
+
+        return tla
+```
+
+The student should be told about buiding lists and join() them.
+
+#### Regular Expressions
+
+Regular Expressions can help to prune the cruft:
+
+```python
+    import re
+    def abbreviate(words):
+        return "".join(item[0].upper() for item in re.findall(r"[a-zA-Z']+", words))
```

Comment:
If we're talking about regex anyway, perhaps an example that avoids the need for indexing?

```suggestion
        return "".join(item.upper() for item in re.findall(r"\b([a-zA-Z])", words))
```

Reply:
Also, this entire code block should probably have it's indentation reduced by 1 level.

Reply:
The regex has not been altered. If an argument can be made for why the current regex is more appropriate than the suggested one, I will resolve this conversation and approve.

Reply:
@jeffdparker let us know. The rest is all _great_ ❤️